src/OmnichannelChatSDK.ts

```diff
@@ -2146,7 +2147,7 @@ class OmnichannelChatSDK {
     }
 
     private useCoreServicesOrgUrlIfNotSet() {
-        if (!this.omnichannelConfig.orgUrl.startsWith(coreServicesOrgUrlPrefix) && this.chatSDKConfig.internalConfig?.createCoreServicesOrgUrlAtRuntime === true) {
+        if (!isCoreServicesOrgUrl(this.omnichannelConfig.orgUrl) && !(this.chatSDKConfig.internalConfig?.createCoreServicesOrgUrlAtRuntime === false)) {
```

Comment:
```suggestion
        if (!isCoreServicesOrgUrl(this.omnichannelConfig.orgUrl) && (this.chatSDKConfig.internalConfig?.createCoreServicesOrgUrlAtRuntime ?? true)) {
```

This is a little confusing to read, so I'm not sure if my suggestion is correct, but maybe it helps.